apps/core-v2-testnet/scripts/deployMockProtocol.ts

```diff
@@ -0,0 +1,189 @@
+  const yvUSDC = await deployMockYearnVault(deployer, usdc.address);
+  const yvDAI = await deployMockYearnVault(deployer, dai.address);
+  const yvWETH = await deployMockYearnVault(deployer, weth.address);
+
+  // Deploy mock yield adapter (term) for each vault token
+
+  const usdcTerm = await deployMockYieldAdapter(
+    deployer,
+    yvUSDC.address,
+    governance.address,
+    linkHash,
+    forwarderFactory.address,
+    usdc.address,
+  );
+
+  const daiTerm = await deployMockYieldAdapter(
+    deployer,
+    yvDAI.address,
+    governance.address,
+    linkHash,
+    forwarderFactory.address,
+    dai.address,
+  );
+
+  const wethTerm = await deployMockYieldAdapter(
+    deployer,
+    yvWETH.address,
+    governance.address,
+    linkHash,
+    forwarderFactory.address,
+    weth.address,
+  );
+
+  // Mint tokens to deployer EOA and give term contract max allowance
+  await usdc.mint(user.address, 1_000_000);
+  await usdc
+    .connect(user)
+    .approve(usdcTerm.address, ethers.constants.MaxUint256);
+
+  await dai.mint(user.address, 1_000_000);
+  await dai.connect(user).approve(daiTerm.address, ethers.constants.MaxUint256);
+
+  // TODO @cashd: use permit here
+  await weth.mint(user.address, 100_000);
+  await weth
+    .connect(user)
+    .approve(wethTerm.address, ethers.constants.MaxUint256);
+
+  // Create 90 and 180 day terms
+  const currentTimestamp = await getCurrentBlockTimestamp(ethers.provider);
+  const start = currentTimestamp + ONE_MINUTE_IN_SECONDS;
+  const expiry90 = currentTimestamp + ONE_DAY_IN_SECONDS * 90;
+  const expiry180 = currentTimestamp + ONE_DAY_IN_SECONDS * 180;
+
+  await createTerm(user, usdcTerm, start, expiry90, user.address, 200_000);
+  await createTerm(user, usdcTerm, start, expiry180, user.address, 200_000);
+  await createTerm(user, daiTerm, start, expiry90, user.address, 200_000);
+  await createTerm(user, daiTerm, start, expiry180, user.address, 200_000);
+  await createTerm(user, wethTerm, start, expiry90, user.address, 15_000);
+  await createTerm(user, wethTerm, start, expiry180, user.address, 15_000);
+
+  // Create pools for every term
+  const fee = 0;
+
+  const usdcPool = await deployMockPool(
+    deployer,
+    usdcTerm,
+    usdc,
+    fee,
+    linkHash,
+    governance,
+    forwarderFactory,
+  );
+
+  const daiPool = await deployMockPool(
+    deployer,
+    daiTerm,
+    dai,
+    fee,
+    linkHash,
+    governance,
+    forwarderFactory,
+  );
+
+  const wethPool = await deployMockPool(
+    deployer,
+    wethTerm,
+    weth,
+    fee,
+    linkHash,
+    governance,
+    forwarderFactory,
+  );
+
+  // give allowance from user to pool
+  await usdc
+    .connect(user)
+    .approve(usdcPool.address, ethers.constants.MaxUint256);
+  await dai.connect(user).approve(daiPool.address, ethers.constants.MaxUint256);
+  await weth
+    .connect(user)
+    .approve(wethPool.address, ethers.constants.MaxUint256);
+
+  // register pool for every 90 / 180 day term
```

Comment:
So you have to create the term first, then register it's id? What does registering do?

Reply:
yes the term needs to exist first because registering a new pool will deposit into the term. Registering a pool is used to configure a liquidity pool with initial values and reserves.